crates/sp1/src/guest.rs

```diff
@@ -0,0 +1,49 @@
+//! This module implements the `ZkvmGuest` trait for the SP1 VM.
+use borsh::{BorshDeserialize, BorshSerialize};
+use sov_rollup_interface::zk::{Zkvm, ZkvmGuest};
+use sp1_zkvm::io;
+
+use crate::VerifyingKey;
+
+/// A guest for the SP1 VM. Implements the `ZkvmGuest` trait
+///  in terms of Risc0's env::read and env::commit functions.
+#[derive(Default)]
+pub struct SP1Guest {}
+
+impl SP1Guest {
+    /// Constructs a new Risc0 Guest
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Zkvm for SP1Guest {
+    type CodeCommitment = VerifyingKey;
+    type Error = anyhow::Error;
+
+    fn verify(
+        _serialized_proof: &[u8],
+        _code_commitment: &Self::CodeCommitment,
+    ) -> Result<Vec<u8>, Self::Error> {
+        unimplemented!()
+    }
+
+    fn verify_and_extract_output<Da: sov_rollup_interface::da::DaSpec, Root: BorshDeserialize>(
+        _serialized_proof: &[u8],
+        _code_commitment: &Self::CodeCommitment,
+    ) -> Result<sov_rollup_interface::zk::StateTransition<Da, Root>, Self::Error> {
+        unimplemented!()
+    }
+}
+
+impl ZkvmGuest for SP1Guest {
+    fn read_from_host<T: BorshDeserialize>(&self) -> T {
+        let buf = io::read_vec();
```

Comment:
btw, checkout how sp1's API for committing / reaading is, we might work around without needing borsh serialize / deserialize. low chance but possible.

also create issue so that we benchmark serialization methods for sp1, i don't think we'll have different results from risc0 but we should try.

Reply:
sp1 does not have reader implementation as risc0 does, so the whole vector should be allocated anyways. the internal io::read::<T>() method uses bincode internally, so we might potentially ditch borsh and just go with the bincode everywhere.

Reply:
ah that's a shame. 

put a todo comment for this in the files pls